### ISSUE TITLE
connectedCallback is not called if a custom element is inserted as part of DocumentFragment

### ISSUE THREAD

**Issue by asurkov** [jsdom/jsdom#3290]
### Basic info:

jsdom 
- **Node.js version: 16 -->
- **jsdom version: starting from 18.0.1/18.1.0 (18.0.0 works) -->

### Minimal reproduction case

```js
import { JSDOM } from 'jsdom';

const { window } = new JSDOM(`<!DOCTYPE html><html><body></body></html>`);

class CustomElement extends window.HTMLElement {
  connectedCallback() {
    console.log('CustomElement connectedCallback');
  }
}

window.customElements.define('custom-element', CustomElement);

class ComplexElement extends window.HTMLElement {
  connectedCallback() {
    let template = window.document.createElement('template');
    template.innerHTML = '<custom-element></custom-element>';
    this.appendChild(window.document.importNode(template.content, true));
    console.log('ComplexElement connectedCallback')
  }
}

window.customElements.define('complex-element', ComplexElement);

window.document.body.appendChild(window.document.createElement('complex-element'));

```
### How does similar code behave in browsers?

CustomElement.connectedCallback is called in the browsers. Also it used to be called in 18.0.0. It stopped working in 18.0.1.

**Comment by asurkov** [comment]
it looks like https://github.com/jsdom/jsdom/commit/1f5acaee40a19ae577df3f50e434acd6f50e1af7 is a culprit

**Comment by domenic** [comment]
Great find. We'd welcome a targeted fix if you or @GrantGryczan are able to find one, along with web platform tests to prevent any regressions. Otherwise I can try to work on reverting that performance change soon (again, I'll add tests at the same time).

**Comment by GrantGryczan** [comment]
Well that sucks, sorry about that. I'll look into it ASAP when I have free time.

**Comment by GrantGryczan** [comment]
I've found a bit of time to look at this tonight, and I've figured out the issue. It's because it sets `isConnected` to `nodeImpl.isConnected` and assumes the same `isConnected` value of all of `nodeImpl`'s descendants, when that is not the case for document fragments. Document fragments can never be connected. Submitting a PR with the fix, which is to check the `isConnected` value of the first child `node` instead of the `nodeImpl` parent.
